app/src/navigation/MainTabNavigator.tsx

```diff
@@ -1,45 +1,92 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
 import { FontAwesome5 } from "@expo/vector-icons";
 import DocumentsStackNavigator from "./DocumentsStackNavigator";
 import ProfileStackNavigator from "./ProfileStackNavigator";
 import VendorsStackNavigator from "./VendorsStackNavigator";
-import { ParamListBase, RouteProp } from "@react-navigation/native";
+import {
+  CommonActions,
+  ParamListBase,
+  RouteProp,
+  useNavigation
+} from "@react-navigation/native";
 import colors from "../styles/colors";
+import { MainTabNavigation } from "../types/navigation";
+import { Linking } from "react-native";
 
 export type MainTabParamList = {
   Profile: undefined;
-  SupportedExchanges: undefined;
+  SupportedExchanges: undefined | { screen: string; params: any };
```

Comment:
Type declarations as undefined? Why pass it then?

Reply:
react navigation quirk